### ISSUE TITLE
Array of array shapes in phpdoc is incorrectly parsed?

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#614]
Using the latest master:

`ERROR: InvalidArgument - src/complex.php:6:25 - Argument 1 of accepts_array_of_shapes expects array{b:bool, d:string, 2:[}, array{b:true} provided`

Expected: An error, but with `array<mixed,array{b:bool, d:string}>`. The serialization or unserialization seems broken from the fragment `2:[}`

```php
<?php
/** @param array{b:bool,d:string}[] $data */
function accepts_array_of_shapes(array $data) {
    var_export($data);
}
accepts_array_of_shapes(['b' => true]);
```

**Comment by muglug** [comment]
Thanks! Broken because I redid parsing to be a bit more clear.
